cmd/kubeadm/app/util/kubeconfig/builder/builder.go

```diff
@@ -0,0 +1,133 @@
+package builder
```

Comment:
Overall, I like the idea here. I think that this can greatly simplify and make the code more understandable.
But I think, that this is outside of the scope of this PR.
Possibly spit it up as a separate PR with the idea of cleaning up the contents and interface of `kubeconfig.go`?